plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -309,7 +309,7 @@ export class GitLabClient {
   ): Promise<PagedResponse<T>> {
     const request = new URL(`${this.config.apiBaseUrl}${endpoint}`);
     for (const key in options) {
-      if (options[key]) {
+      if (options[key]?.toString()) {
```

Comment:
```suggestion
      if (options[key] !== undefined) {
```

I believe it's a bit clearer to check for not undefined instead.

Reply:
Will test it tomorrow morning, just to be sure it works still. But yeah, that makes it a bit clearer

Reply:
I came up with a different solution. If checking for undefined, an empty string would be allowed. So I'm checking it's not undefined and neither an empty string. I added some tests as well that check that behavior